internal/blocksync/pool.go

```diff
@@ -376,13 +385,15 @@ func (pool *BlockPool) RemovePeer(peerID types.NodeID) {
 	pool.mtx.Lock()
 	defer pool.mtx.Unlock()
 
-	pool.removePeer(peerID)
+	pool.removePeer(peerID, true)
 }
 
-func (pool *BlockPool) removePeer(peerID types.NodeID) {
-	for _, requester := range pool.requesters {
-		if requester.getPeerID() == peerID {
-			requester.redo(peerID)
+func (pool *BlockPool) removePeer(peerID types.NodeID, redo bool) {
```

Comment:
Can you explain this redo concept and why it is needed?

Reply:
This is actually a merge from the main branch to fix a peer test.